spec/models/topic_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Topic, type: :model do
+  subject { create(:topic) }
+
+  describe 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_uniqueness_of(:name) }
+  end
+
+  describe '#create' do
+    it 'creates the topic' do
+      expect { subject }.to change { Topic.count }.by(1)
+    end
+
+    it 'has the given name' do
+      create(:topic, name: 'Sports')
+      expect(Topic.last.name).to eq('Sports')
+    end
```

Comment:
I would recommend to start using the rspec json_expectations. This gem allows you to check the entire json response, so that you don't have to check each of the properties in a `it` block. if you have any questions. Here is the [link to a commit](https://github.com/rootstrap/ruby-openform-api/pull/25/commits/a8a68e41aa175dae1c64d90aabd17f13d4f22ada) where it was added. Basically you have to add it to your `gemfile` and in the `spec_helper.rb`.

Reply:
Thank you, i will check and add these gems.

Reply:
I will add these gems later, because now i am not testing the json, i am testing the given name is the expected. That will be really helpful when i test the end pints.